### PR TITLE
Embedding data roles in ciphertext

### PR DESCRIPTION
**Pull request by DeirdreCleary**
# Why

Data roles need to be embedded in the ciphertext to support data policies

# How

Embedding the data role in the ciphertext
Performing KDF on the new version with `Secp256k1` curve
Adding curve parameters for `Secp256k1` curve

### CODE REVIEW

File: evervault/crypto/client.py [hunk 1]
```diff
@@ -37,87 +39,135 @@ def __init__(self, api_key=None, curve="SECP256K1", max_file_size_in_mb=25):
...
+            metadata = self.__generate_metadata(metadata["role"])
+            metadata_offset = struct.pack(
+                "<H", len(metadata)
+            )  # '<H' specifies 16-bit unsigned little-endian
+            payload = metadata_offset + metadata + bytes(coerced_data, "utf8")
+        else:
+            payload = bytes(coerced_data, "utf8")
 
         encrypted_bytes = b""
         if self.curve == SECP256K1:
```
Comment (jakekgrog) [inline evervault/crypto/client.py:129]: This is causing the K1 failure for metadata:
```suggestion
        if self.curve == SECP256K1 and not has_role:
```

Comment (DeirdreCleary) [reply evervault/crypto/client.py:129]: Legend 👑

File: evervault/client.py [hunk 2]
```diff
@@ -43,8 +43,8 @@ def __init__(
     def _auth(self):
         return (self.api_key, "")
 
-    def encrypt(self, data):
-        return self.crypto_client.encrypt_data(self, data)
+    def encrypt(self, data, metadata):
```
[the remark this answers is not in the document]
Comment (DeirdreCleary) [reply evervault/client.py:46]: Oh grand, I'd had that and then switched it out, will switch back!

**Review by e-moran — APPROVED** [review]
Lgtm! But I'm not totally familiar with how we've implemented it in the other SDKs - so might be worth getting someone else to take a look too.

**Review by jakekgrog — APPROVED** [review]

**Pull request merged**